test/unit/math/prim/fun/lgamma_stirling_diff_test.cpp

```diff
@@ -0,0 +1,110 @@
+#include <stan/math/prim.hpp>
+#include <gtest/gtest.h>
+#include <algorithm>
+#include <limits>
+#include <vector>
+#include <cmath>
+
+TEST(MathFunctions, lgamma_stirling_diff_errors_special_cases) {
+  using stan::math::lgamma_stirling_diff;
+
+  double nan = std::numeric_limits<double>::quiet_NaN();
+  double inf = std::numeric_limits<double>::infinity();
+
+  EXPECT_TRUE(std::isnan(lgamma_stirling_diff(nan)));
+  EXPECT_FLOAT_EQ(lgamma_stirling_diff(inf), 0);
+  EXPECT_THROW(std::isnan(lgamma_stirling_diff(-1.0)), std::domain_error);
+  EXPECT_FLOAT_EQ(lgamma_stirling_diff(0.0), inf);
+}
+
+TEST(MathFunctions, lgamma_stirling_diff_accuracy) {
+  using stan::math::lgamma_stirling_diff;
+  using stan::math::lgamma_stirling_diff_useful;
+
+  double start = std::nextafter(10, 11);
+  for (double x = start; x < 1e8; x *= 1.5) {
+    long double x_l = static_cast<long double>(x);
+    long double stirling
+        = 0.5 * std::log(2 * static_cast<long double>(stan::math::pi()))
+          + (x_l - 0.5) * std::log(x_l) - x_l;
+    long double lgamma_res = std::lgamma(x_l);
+    double diff_actual = static_cast<double>(lgamma_res - stirling);
+    double diff = lgamma_stirling_diff(x);
+    double tol = std::max(1e-06 * 0.5 * (fabs(diff_actual) + fabs(diff)), 1e-6);
+
+    EXPECT_NEAR(diff, diff_actual, tol)
+        << "x = " << x << "; lgamma = " << lgamma_res
+        << "; stirling = " << stirling;
+  }
+
+  double before_useful = std::nextafter(lgamma_stirling_diff_useful, 0);
+  double after_useful = std::nextafter(lgamma_stirling_diff_useful,
+                                       stan::math::positive_infinity());
+
+  double value_before = lgamma_stirling_diff(before_useful);
+  double value_at = lgamma_stirling_diff(lgamma_stirling_diff_useful);
+  double value_after = lgamma_stirling_diff(after_useful);
+  double diff_before = value_at - value_before;
+  double diff_after = value_after - value_at;
+  double tol
+      = std::max(1e-14 * (0.5 * (fabs(diff_before) + fabs(diff_after))), 1e-14);
+  EXPECT_NEAR(diff_before, diff_after, tol) << "diff around useful cutoff";
+}
+
+namespace lgamma_stirling_diff_test_internal {
```

Comment:
There's no need to namespace this in a test file.

Reply:
I've heard there is a plan to run all unit tests in  a single translation unit to save on compilation time (https://discourse.mc-stan.org/t/requiring-global-unique-names-for-unit-tests-in-stan-math/11270). Since I do have `TestValue` and `testValues` in other tests, this seemed like a reasonable precaution.